src/pages/index.js

```diff
@@ -32,79 +39,167 @@ const profileDescriptionInput = document.querySelector(
+/*                               Form Validators                              */
+/* -------------------------------------------------------------------------- */
 
 const profileFormValidator = new FormValidator(settings, profileEditModal);
 profileFormValidator.enableValidation();
 
 const addFormValidator = new FormValidator(settings, addCardModal);
 addFormValidator.enableValidation();
 
-// Event listeners
-profileEditButton.addEventListener("click", () => {
-  const user = userInfo.getUserInfo();
-  profileTitleInput.value = user.name;
-  profileDescriptionInput.value = user.description;
-  modalEditForm.open();
-});
+const editProfilePicFormValidator = new FormValidator(
+  settings,
+  editProfileImageModal
+);
+editProfilePicFormValidator.enableValidation();
 
-addNewCardButton.addEventListener("click", () => {
-  addFormValidator._disableSubmitButton();
-  addCardImageModal.open();
+/* -------------------------------------------------------------------------- */
+/*                        Class Insantiation Constants                        */
+/* -------------------------------------------------------------------------- */
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "675ffc74-7010-43d4-a4e4-01622ad8dbf1",
+    "Content-Type": "application/json",
+  },
 });
 
-function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+/* -------------------------------------------------------------------------- */
+/*                            Profile Picture Edit                            */
+/* -------------------------------------------------------------------------- */
 
-  return card.getView();
+function loadInitialProfilePicture() {
+  api
+    .loadInitialProfileData()
+    .then((profileData) => {
+      profilePictureImage.src = profileData.avatar;
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+}
+
+loadInitialProfilePicture();
+
+/* -------------------------------------------------------------------------- */
+/*                               Event Hanlders                               */
+/* -------------------------------------------------------------------------- */
+
+function handleEditProfileSubmit(formData) {
+  modalEditForm.setSubmitText(true);
+  api
+    .editProfileData(formData)
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+      userInfo.setUserInfo(data);
+      modalEditForm.close();
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => modalEditForm.setSubmitText(false));
 }
 
-function handleEditProfileSubmit(data) {
-  userInfo.setUserInfo(data);
-  modalEditForm.close();
+function handleEditProfilePictureSubmit(data) {
+  profilePictureImage.src = data.url;
```

Comment:
`  profilePictureImage.src = data.url;
`

You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change